NeoDebuggerUI/ViewModels/MainWindowViewModel.cs

```diff
@@ -226,20 +227,70 @@ public async Task OpenRunDialog()
 
             if (!IsSteppingOrOnBreakpoint)
             {
-                var task = modalWindow.ShowDialog();
+                var task = modalWindow.ShowDialog(new Window());
                 await Task.Run(() => task.Wait());
             }
 
             // not using getters because the properties are updated on another thread and won't update the ui
             IsSteppingOrOnBreakpoint = DebuggerStore.instance.manager.IsSteppingOrOnBreakpoint;
             ConsumedGas = DebuggerStore.instance.UsedGasCost;
 
+            EvtDebugCurrentLineChanged?.Invoke(IsSteppingOrOnBreakpoint, DebuggerStore.instance.manager.CurrentLine + 1);
             if (IsSteppingOrOnBreakpoint)
             {
                 UpdateStackPanel();
             }
         }
 
+        public void AddBreakpoint(int line)
+        {
+            DebuggerStore.instance.manager.AddBreakpoint(line - 1, SelectedFile);
+        }
+
+        public void RemoveBreakpoint(int line)
+        {
+            DebuggerStore.instance.manager.RemoveBreakpoint(line - 1, SelectedFile);
+        }
+
+        public void SetBreakpoint(int line)
+        {
+            if (Breakpoints.Contains(line))
+            {
+                // remove breakpoint
+                RemoveBreakpoint(line);
+                EvtBreakpointStateChanged?.Invoke(line, false);
+                return;
```

Comment:
I think you should use if-else instead of many returns on this method.
A better use of many returns is when you want to stop the continuation of a method that have a single path, not a method that can take many directions, usually for validation